src/framework/components/screen/component.js

```diff
@@ -129,97 +131,66 @@ pc.extend(pc, function () {
-                    // we have to compue clip-space-to-camera-space transform ourself
-
-                    // we extract fov from the camera
-                    var fov = camera.fov / 2;
-                    // then we compute the viewport height at nearClipOffset distance which is conveniently
-                    // fov angle tangents times offset of the place
-                    var nearClipHalfHeight = Math.tan( fov * Math.PI / 180.0 ) * Math.abs( nearClipOffset );
-                    // the near clip width comes from screen proportions
-                    var nearClipHalfWidth  = nearClipHalfHeight * w / h;
-                    
-                    // while the clip space to near clip space would be just scale
-                    clipSpaceToNearClipSpace.setTRS( pc.Vec3.ZERO, pc.Quat.IDENTITY, new pc.Vec3( nearClipHalfWidth, nearClipHalfHeight, 1 ) );
+            // do we follow the camera?
+            if ( screenType == pc.SCREEN_TYPE_CAMERA ) {
+                if ( camera.projection == pc.PROJECTION_PERSPECTIVE ) {
+                    this._planeHeight = Math.tan( camera.fov / 2.0 * Math.PI / 180.0 ) * Math.abs( 2 * this._screenDistance );
                 } else {
-                    // we are in ortho camera
-                    clipSpaceToNearClipSpace.setTRS( pc.Vec3.ZERO, pc.Quat.IDENTITY, new pc.Vec3( camera.orthoHeight * w / h, camera.orthoHeight, 1 ) );    
+                    this._planeHeight = camera.orthoHeight * 2.0;
                 }
 
-                // the clipOffset will be the transform to move from (0, 0, 0) onto the desired UI pane
-                var clipOffset = new pc.Mat4().setTRS( new pc.Vec3(0, 0, -nearClipOffset), pc.Quat.IDENTITY, pc.Vec3.ONE );
-
-                // and the screen matrix is effectively the chain of transforms:
-                // UI plane -> Clip space -> Camera Origin -> Camera plane
-                this._screenMatrix = camera._node.getWorldTransform().clone().
-                    mul( clipOffset ).
-                    mul( clipSpaceToNearClipSpace ).
-                    mul( this._screenMatrix );
-            } else if (this._screenType == pc.SCREEN_TYPE_WORLD) {
-                w = this._resolution.x;
-                h = this._resolution.y;
-
-                // in case of the the world everything is very simple – just normalize the size to match
-                // the desired "resolution"
-                var worldMatrix = new pc.Mat4();
-                //worldMatrix.setTRS( new pc.Vec3( 0, 0, 0 ), pc.Quat.IDENTITY, new pc.Vec3( 1, 1, 1 ) );
-                worldMatrix.setTRS( new pc.Vec3( this._offset.x, this._offset.y, 0 ), pc.Quat.IDENTITY, new pc.Vec3( 1, 1, 1 ) );
-
-                this._screenMatrix = worldMatrix; 
+                // compose a transform to convert screen points into world space onto the screen plane (whether it's ortho or perspective)
+                this._screenMatrix.setIdentity();
```

Comment:
Should it be an identity? The comment states that we compose a transform...

Reply:
The comment is wrong, gonna amend now